frontend/__tests__/inquiry/add_inquiry.test.tsx

```diff
@@ -0,0 +1,94 @@
+/* Codacy is unhappy with await */
+/* eslint-disable @typescript-eslint/await-thenable */
+/* Codacy is unhappy with click events */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* Codacy is unhappy with mocking */
+/* eslint-disable @typescript-eslint/no-empty-function */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import { fireEvent, render, screen } from "@testing-library/react"
+import userEvent from "@testing-library/user-event"
+import { Inquiries } from "../../src/routes/_layout/inquiries"
+import "@testing-library/jest-dom"
+import {
+  MAX_INQUIRY_LENGTH,
+  MIN_INQUIRY_LENGTH,
+} from "../../src/components/Inquiries/AddInquiry"
+
+const unicodeText =
+  "Тенденция к взаимопомощи у человека имеет столь отдаленное происхождение и так глубоко переплетена со всей прошлой эволюцией человеческого рода, что она сохранилась у человечества вплоть до настоящего времени, несмотря на все превратности истории."
+const nonUnicodeText =
+  "\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF"
+
+jest.mock("../../src/components/Inquiries/InquiriesTable", () => ({
+  __esModule: true,
+  default: () => <div />,
+}))
+
+jest.mock("@tanstack/react-query", () => ({
+  ...jest.requireActual("@tanstack/react-query"),
+  useQueryClient: () => {},
+  useMutation: () => ({
+    mutate: () => {},
+  }),
+}))
+
+describe("Add Inquiry", () => {
+  beforeEach(async () => {
+    render(<Inquiries />)
+    await userEvent.click(screen.getByText("Add Inquiry"))
+  })
+
+  it("should display add modal when user presses Add Inquiry button", async () => {
+    const textArea = await screen.getByTestId("add-inquiry-text")
+    fireEvent.change(textArea, {
+      target: {
+        value: "Why do birds suddenly appear every time you are near?",
+      },
+    })
+    await userEvent.click(screen.getByTestId("submit-add-inquiry"))
+  })
+
+  it("should display required error when no string is entered", async () => {
+    await userEvent.click(screen.getByTestId("submit-add-inquiry"))
+    await screen.getByText("Inquiry text is required.")
```

Comment:
so what happens if the error isn't there? Is an exception thrown?

Reply:
Essentially yes.  After 3000ms (I think) the assertion errors out.  There are a few different assertions packed into this line -- that the HTML Element exists, that it is clickable and so on

Reply:
```bash
d display error message when user enters inquiry more than 255 characters (87 ms)
    ✓ should display error message when user enters a non-Unicode string (88 ms)
    ✓ should accept foreign language unicode characters (80 ms)

  ● Add Inquiry › should display required error when no string is entered

    TestingLibraryElementError: Unable to find an element with the text: Inquiry text is requiry.. This could be be
cause the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.

    Ignored nodes: comments, script, style
    <body
      data-scroll-locked=""
    >
      <div
        aria-hidden="true"
        data-aria-hidden="true"
      >
        <div
          class="chakra-container css-raz39d"
        >
          <h2
            class="chakra-heading css-9yq14l"
          >
            Inquiries Management
          </h2>
          <div
            class="css-wxfe5d"
          >
            <button
              class="chakra-button css-1279lbq"
              type="button"
            >
              <svg
                class="chakra-icon css-13otjrl"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32
 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                />
              </svg>
               Add
              Inquiry
            </button>
          </div>
          <div />
        </div>
      </div>
      <div
        class="chakra-portal"
      >
        <div
          aria-hidden="true"
          class="chakra-modal__overlay css-13d1tkw"
          data-aria-hidden="true"
          style="opacity: 0.5551976485003252;"
        />
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"   
          tabindex="0"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-1oxhx2p"
            tabindex="-1"
          >
            <form
              animate="enter"
              aria-describedby="chakra-modal--body-:r6:"
              aria-labelledby="chakra-modal--header-:r6:"
              aria-modal="true"
              class="chakra-modal__content css-6yiygt"
              custom="[object Object]"
              exit="exit"
              id="chakra-modal-:r6:"
              initial="exit"
              role="dialog"
              tabindex="-1"
              variants="[object Object]"
            >
              <header
                class="chakra-modal__header css-72fd9l"
                id="add-inquiry-show-modal"
              >
                Add Inquiry
              </header>
              <button
                aria-label="Close"
                class="chakra-modal__close-btn css-6sdaxy"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,
1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="chakra-modal__body css-1gdm77d"
                id="chakra-modal--body-:r6:"
              >
                <div
                  class="chakra-form-control css-0"
                  data-focus=""
                  data-invalid=""
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    data-focus=""
                    data-invalid=""
                    for="text"
                    id="field-:r9:-label"
                  >
                    Inquiry Text
                  </label>
                  <textarea
                    aria-describedby="field-:r9:-feedback"
                    aria-invalid="true"
                    class="chakra-textarea css-0"
                    data-testid="add-inquiry-text"
                    id="text"
                    name="text"
                    placeholder="Enter the text of your inquiry."
                  />
                  <div
                    aria-live="polite"
                    class="chakra-form__error-message css-70qvj9"
                    id="field-:r9:-feedback"
                  >
                    Inquiry text is required.
                  </div>
                </div>
              </div>
              <footer
                class="chakra-modal__footer css-1q2vdcx"
              >
                <button
                  class="chakra-button css-4xx2wk"
                  data-testid="submit-add-inquiry"
                  type="submit"
                >
                  Save
                </button>
                <button
                  class="chakra-button css-4xx2wk"
                  type="button"
                >
                  Cancel
                </button>
              </footer>
            </form>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"   
          tabindex="0"
        />
      </div>
    </body>

      53 |   it("should display required error when no string is entered", async () => {
      54 |     await userEvent.click(screen.getByTestId("submit-add-inquiry"))
    > 55 |     await screen.getByText("Inquiry text is requiry.")
         |                  ^
      56 |   })
      57 |
      58 |   it("should display error message when user enters inquiry less than 10 characters", async () => {     

      at Object.getElementError (node_modules/@testing-library/dom/dist/config.js:37:19)
      at node_modules/@testing-library/dom/dist/query-helpers.js:76:38
      at node_modules/@testing-library/dom/dist/query-helpers.js:52:17
      at node_modules/@testing-library/dom/dist/query-helpers.js:95:19
      at Object.<anonymous> (__tests__/inquiry/add_inquiry.test.tsx:55:18)
```